tests/dataset/dond/test_do1d.py

```diff
@@ -313,19 +313,15 @@ def test_do1d_break_condition(caplog: LogCaptureFixture, _param_set, _param) ->
     def break_condition():
         return True
 
-    data = do1d(
-        _param_set,
-        start,
-        stop,
-        num_points,
-        delay,
-        _param,
-        break_condition=break_condition,
-    )
-
-    assert isinstance(data[0], DataSet) is True
-    assert (
-        "qcodes.dataset.dond.do_nd_utils",
-        logging.WARNING,
-        "Measurement has been interrupted, data may be incomplete: Break condition was met.",
-    ) in caplog.record_tuples
+    with pytest.raises(BreakConditionInterrupt, match="Break condition was met."):
+        do1d(
+            _param_set,
+            start,
+            stop,
+            num_points,
+            delay,
+            _param,
+            break_condition=break_condition,
+        )
+
+    assert "Measurement has been interrupted, data may be incomplete" in caplog.text
```

Comment:
That you no longer get a partial dataset back if you interrupt the measurement feels like a breaking change that we should be clear about and at least document

Reply:
thats a good point, is this a breaking change we are okay with or should we preserve the data that was captured before the interrupt?

Reply:
I believe my latest push makes only keyboard interrupts that result in no partial datasets created. Does this sound fair? I'll update the docs and notes for upcoming change logs accordingly if that sounds good.

Reply:
@trevormorgan I think this is nearly ready to go. Could you add a changelog fragment. There also seems to be some test errors on windows that needs to be sorted

Reply:
@jenshnielsen I'll get the fragement added and resolve the test errors today. Thanks for the review!